avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerDe.java

```diff
@@ -86,14 +90,29 @@ protected void configureClientProperties(AbstractKafkaAvroSerDeConfig config) {
   /**
    * Get the subject name for the given topic and value type.
    */
-  protected String getSubjectName(String topic, boolean isKey, Object value) {
-    if (isKey) {
-      return keySubjectNameStrategy.getSubjectName(topic, isKey, value);
+  protected String getSubjectName(String topic, boolean isKey, Object value, Schema schema) {
+    Object subjectNameStrategy = subjectNameStrategy(isKey);
+    if (deprecatedSubjectNameStrategy(subjectNameStrategy)) {
```

Comment:
Why not just the following instead of the `deprecatedSubjectNameStrategy(...)` method:
```
if (subjectNameStrategy instanceof io.confluent.kafka.serializers.subject.v2.SubjectNameStrategy) {
    return ((io.confluent.kafka.serializers.subject.v2.SubjectNameStrategy) subjectNameStrategy)         .getSubjectName(topic, isKey, schema);
} else {
      return ((SubjectNameStrategy) subjectNameStrategy).getSubjectName(topic, isKey, value);
}
```

Also, I reversed the order. Since the newer SubjectNameStrategy is faster, we should check that in case the implementation implements both interfaces.